### DISCUSSION TITLE
TO EVERY USER - Will be interesting to provide a performance assessment of this system

### DISCUSSION THREAD

**Discussion by luy710** [frangoteam/FUXA#982] [General]
Are there any requirements for server hardware, CPU, and memory for this system? What is the maximum number of data collection points it supports? Can provide a performance assessment of this system? Thank you.

**Comment by unocelli** [comment]
Hi, depends since you can also install it on a raspberry...we count on the experiences of users

**Comment by robbudge** [comment]
As with Codesys for PLC, it can be installed on various platforms.That being said try to deploy a Codesys  project with 10,000 Tags Shared  Via  OPCua and 40 TCP devices on  a RPI.
I see mainly high memory usage with  Fuxa, but it depends on the tag count, graphical objects, alarms, Data-Logging etc....
you just have to monitor htop and

**Comment by unocelli** [comment]
You should consider that all graphics run on the client's browser, which is normally not where fuxa runs

**Reply by robbudge** [reply]
Most  times i have a  local screen running chromium, then remote via Web.
I am currently testing a 10" Raspberry PI HMI from ComFile.
https://comfiletech.com/linux-panel-pc/

**Comment by luy710** [comment]
I tested 10000 OPCUA connections at different points
Unable to display value when viewing connection point value
Backend service error
2023-11-25T03:15:38.563Z [ERR]'ns=2; S=Test Test.1616 '_ StartMonitor Error: expecting a valid session here
2023-11-25T03:21:39.501Z [WAR] daqstorage. data overload D_ 6bc85e1e-811e4096 '
2023-11-25T03:22:08.510Z [WAR] daqstorage. data overload! ' D_ 6bc85e1e-811e4096 '
GET/assets/lib/svg/svg. min.js.map 404 5.713 ms -168
2023-11-25T03:22:28.707Z [WAR] daqstorage. data overload! ' D_ 6bc85e1e-811e4096 '
2023-11-25T03:22:32.507Z [WAR] daqstorage. data overload! ' D_ 6bc85e1e-811e4096 '
POST/API/Heartbeat 200 4.037 ms --
2023-11-25T03:22:48.709Z [WAR] daqstorage. data overload D_ 6bc85e1e-811e4096 '
2023-11-25T03:22:52.506Z [WAR] daqstorage. data overload! ' D_ 6bc85e1e-811e4096 '
2023-11-25T03:22:56.508Z [WAR] daqstorage. data overload! ' D_ 6bc85e1e-811e4096 '
GET/API/project 304 97.772 ms --
2023-11-25T03:23:08.959Z [INF] 'FUXA' stop
2023-11-25T03:23:08.972Z [INF] 'FUXA' start
POST/api/projectData 200 274.064 ms --
GET/API/project 304 101.527 ms --
2023-11-25T03:23:13.921Z [INF] 'FUXA' stop
2023-11-25T03:23:13.929Z [INF] 'FUXA' start
POST/api/projectData 200 127.259 ms --
2023-11-25T03:23:22.262Z [WAR] daqstorage. data overload! ' D_ 6bc85e1e-811e4096 '
2023-11-25T03:23:25.836Z [WAR] daqstorage. data overload D_ 6bc85e1e-811e4096 '
Final program exit

**Reply by robbudge** [reply]
I recently had the same on a unit running Codesys Locally. 
it looks likes it’s a local resource issue. Not sure at what end of the connection. 
the shared unit was regularly hitting over 70% utilization with cores regularly hitting 100%
Also the OPCua server would regularly lockup. 

But…. 
I moved my Fuxa project to separate hardware. 
and my OPCua issues disappeared. 
I did stop the Docker running Fuxa on the shared unit. 
the shared unit just running Codesys is now very stable and I have not had any more OPCua Server lockup. 

The separate hardware (RPI) tuning Fuxa directly is quick and responsive and does not appear to be experiencing any of the lags of the shared unit. 

I would certainly suggest testing the project on some hardware with power, just to see if it’s a resource issue. 

Subscribing to that many channels and accepting the data is going to take some processing power. 


I would also disable Daq or at least switch to a remote InfluxDB. 
Local SQL again is a resource hog.

**Reply by luy710** [reply]
NPM start running directly on the server without using Docker
Server configuration with 64 cores CPU and 128 memory
![image](https://github.com/frangoteam/FUXA/assets/19400252/8e170557-85da-407e-86e7-aebeda272d7c)
![image](https://github.com/frangoteam/FUXA/assets/19400252/5da3b489-1bef-424e-9881-0ac13e8176ce)
Final program exit

**Reply by luy710** [reply]
![image](https://github.com/frangoteam/FUXA/assets/19400252/f8c0062b-7042-4479-be05-bd7edc67e038)
Final program exit

**Comment by luy710** [comment]
Can a Node.js backend program only use one CPU core? Is the cluster module not used to create multiple worker processes?"
